### ISSUE TITLE
--write mode creates problematic yaml output with multiline strings

### ISSUE THREAD

**Issue by bnogaro** [ansible/ansible-lint#3583]
##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
ansible-lint 6.17.1 using ansible-core:2.15.1 ruamel-yaml:0.17.26 ruamel-yaml-clib:0.2.7
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

As a mean of complying with the yaml[line-length] rule with long strings without spaces (URL, for instance), one could do the following:
```yaml
---
my_var: "
  Lorem_ipsum_dolor_sit_amet__consectetur_adipiscing_elit__Mauris_vitae_consectetur_lacus__ac_sagittis_metus__Curabitur_\
  nibh_diam__molestie_et_odio_a__volutpat_lacinia_nisi__Mauris_consequat_mi_felis__rutrum_sodales_sapien_varius_placerat
  "
```

This file passes all ansible-lint tests. However, running `ansible-lint --write` reassembles the two lines into one
This exceeds ansible-lint yaml[line-length] criterium, litterally meaning that ansible-lint --write created a ansible-lint violation.

```console
                Rule Violation Summary                
 count tag               profile rule associated tags 
     1 yaml[line-length] basic   formatting, yaml    
```

##### Desired Behavior

Hopefully, ansible-lint --write would preserve the line-length rule, not removing the escaped newlines

##### Actual Behavior

The variable is rewritten on a single line, failing the test.

```yaml
---
my_var: "Lorem_ipsum_dolor_sit_amet__consectetur_adipiscing_elit__Mauris_vitae_consectetur_lacus__ac_sagittis_metus__Curabitur_nibh_diam__molestie_et_odio_a__volutpat_lacinia_nisi__Mauris_consequat_mi_felis__rutrum_sodales_sapien_varius_placerat"
```

**Comment by ajinkyau** [comment]
@bnogaro we will be happy to accept a PR for it.

**Comment by bnogaro** [comment]
Before any contribution, I may need the input from a more experienced developer on this project.
As far as I understant the problem, this seems to come from the 'round-trip' functionality of ruamel.yaml, which doesn't seem to customisable in terms of max length of line in the output.
In your opinion, should it be a contribution to ruamel.yaml, or should we process the ruamel.yaml output with another tool to break long lines into multiline strings, using the yamllint configured max line length?

**Comment by shatakshiiii** [comment]
Hey @bnogaro, ruamel.yaml is not able to preserve these escaped newlines right now. 
Ruamel ticket for this issue: https://sourceforge.net/p/ruamel-yaml/tickets/474/

However, ruamel does support the other recommended syntax for writing multiline strings (i.e. using Block Style Indicators "|" and ">"). We are documenting this thing over [here](https://ansible.readthedocs.io/projects/lint/rules/yaml/#additional-information-for-multiline-strings), using which `ansible-lint --write`  will no longer collapse your multiline string.
